drivers/serial/Kconfig.ns16550

```diff
@@ -103,6 +103,17 @@ config UART_NS16550_WA_ISR_REENABLE_INTERRUPT
 	  the IER is being toggled to re-assert interrupts at the end of ISR
 	  to nudge the host interrupt controller to fire the ISR again.
 
+config UART_NS16550_WA_TX_FIFO_EMPTY_INTERRUPT
+	bool "Callback directly when the TX FIFO is already empty"
+	default y
```

Comment:
This should not default to be enabled, and should only be enabled when needed.

Reply:
You can add a condition to enable by default if the shell UART backend is enabled.